### PR TITLE
[agent-b][issue #704] Add agent conversation read owner

### LINKED ISSUE #704: [api] Implement v1 agent/conversation read-model owner
## Summary

Child of #665 and #694 for the next Slice 6 read-model owner family after:

- #696 / PR #697 closed persisted observability reads: `run.trace`, `event.list`, `event.get`, `runtime.logs`, `runtime.incidents`.
- #700 / PR #701 closed entity-native reads: `entity.list`, `entity.get`, `entity.aggregate`.

This issue covers the remaining agent/conversation read-model family:

- `agent.list`
- `agent.get`
- `conversation.list`
- `conversation.get`
- `conversation.current_for_agent`

## Category

High-risk operator-surface / runtime read-model parity and semantic ownership work.

## Required pre-audit before coding

Start with a `Pre-Implementation Coverage Audit` and independent gate. Do not implement until the gate approves the class boundary.

The audit must identify the canonical owner across:

- agent declarations
- sessions
- turns
- conversation audits
- pending deliveries
- lifecycle facts

It must classify dashboard `SQLAgentReader` / `SQLConversationReader` as one of:

- canonical owner
- adapter consuming the owner
- bypass / same-concept interpreter that must move or be split/escalated

It must prove no same-concept v1/dashboard/Builder/CLI interpreter remains hidden.

## Governing spec sections

Use the authoritative platform spec:

- `docs/specs/swarm-platform/platform/contracts/platform-spec.yaml` `api_specification.method_catalog.agent.list`
- `docs/specs/swarm-platform/platform/contracts/platform-spec.yaml` `api_specification.method_catalog.agent.get`
- `docs/specs/swarm-platform/platform/contracts/platform-spec.yaml` `api_specification.method_catalog.conversation.list`
- `docs/specs/swarm-platform/platform/contracts/platform-spec.yaml` `api_specification.method_catalog.conversation.get`
- `docs/specs/swarm-platform/platform/contracts/platform-spec.yaml` `api_specification.method_catalog.conversation.current_for_agent`
- Relevant `components.schemas` for agent and conversation read results.

If any exact schema/method section is missing, stale, or insufficient, stop and repair/escalate the spec before implementation.

## Non-closure boundary

This child may close only the agent/conversation read-model owner slice. It must not claim closure of full #665 Slice 6, subscriptions, agent mutations, CLI migration, or #677/#683 control tail unless the pre-audit widens the class and the lead explicitly approves that broader scope.

Parent: #694
Part of #665

### PR DESCRIPTION
**Pull request by yazzaoui**
## Summary
- Add a shared `internal/store` agent/conversation operator read owner for `agent.list`, `agent.get`, `conversation.list`, `conversation.get`, and `conversation.current_for_agent`.
- Wire v1 JSON-RPC handlers to that owner with typed `AGENT_NOT_FOUND` / `SESSION_NOT_FOUND` behavior and conversation filter/cursor parsing.
- Demote dashboard `/api/agents*` and `/api/conversations*` to adapters over the shared owner; dashboard endpoint files no longer contain direct agent/conversation SQL ownership.

Closes #704
Part of #665
Part of #694

## Governing refs/context
- `docs/specs/swarm-platform/platform/contracts/platform-spec.yaml` `api_specification.method_catalog.agent.list`
- `docs/specs/swarm-platform/platform/contracts/platform-spec.yaml` `api_specification.method_catalog.agent.get`
- `docs/specs/swarm-platform/platform/contracts/platform-spec.yaml` `api_specification.method_catalog.conversation.list`
- `docs/specs/swarm-platform/platform/contracts/platform-spec.yaml` `api_specification.method_catalog.conversation.get`
- `docs/specs/swarm-platform/platform/contracts/platform-spec.yaml` `api_specification.method_catalog.conversation.current_for_agent`
- `components.schemas.AgentSummary`, `AgentDetail`, `ConversationSummary`, `ConversationDetail`
- #704 pre-audit and approved gate comment

## Semantic migration accounting
- New canonical owner: `internal/store/operator_agent_conversation_read_surface.go`.
- Old non-authoritative producer/reader path invalidated: dashboard-local agent/conversation endpoint projection ownership in `internal/dashboard/server/agents_sql.go`, `internal/dashboard/server/conversations_sql.go`, and fallback handler scans.
- Production paths checked for surviving old behavior: v1 handler registry, dashboard `/api/agents*`, dashboard `/api/conversations*`, Builder/CLI/script reads, and dashboard SQL reader files.
- Migration complete for #704: yes, for the approved agent/conversation read-model child only. Subscriptions, agent control/mutation, CLI migration, and #677/#683 control tail remain outside this closure claim.

## Tests
- `go run ./cmd/swarm-openrpc-gen --check`
- `go test ./internal/store ./internal/apiv1 ./internal/dashboard/server -count=1`
- `go test ./... -count=1`

### CODE REVIEW

**Comment by yazzaoui** [comment]
## Post-Implementation Proof Audit

Current head SHA: `61c49c5bcbd16cb46c7d8dbe88f1123b443162aa`

Exact semantic concepts changed:
- Agent operator read-model truth for `agent.list` and `agent.get`.
- Conversation operator read-model truth for `conversation.list`, `conversation.get`, and `conversation.current_for_agent`.
- Dashboard REST agent/conversation reads as deprecated adapters over the same owner.

Chosen working failure class closed by this PR: missing canonical agent/conversation read-model owner, with v1 methods unavailable and dashboard SQL readers acting as same-concept interpreters.

Broader failure class: #694 / #665 operator-facing read-model completion.

Parent failure class above it: #665 user-facing RPC API implementation track. Parent remains open for subscriptions/Slice 7, CLI migration, agent control/mutation, and #677/#683 control tail.

Canonical owner after change:
- `internal/store/operator_agent_conversation_read_surface.go`
- Owner methods: `ListOperatorAgents`, `LoadOperatorAgent`, `ListOperatorConversations`, `LoadOperatorConversation`, `LoadCurrentOperatorConversationForAgent`.
- Compatibility methods for deprecated dashboard DTOs remain in the same store owner: `ListDashboardOperatorConversations`, `LoadDashboardOperatorConversation`.

Touched consumers of the owner:
- v1 JSON-RPC handlers in `internal/apiv1/operator_read.go`.
- `cmd/swarm/main.go` v1 wiring through `AgentConversations: stores.Postgres`.
- Dashboard `/api/agents`, `/api/agents/{id}`, `/api/conversations`, `/api/conversations/{sessionID}` through canonical reader interfaces in `internal/dashboard/server/server.go`.
- Dashboard `SQLAgentReader` / `SQLConversationReader` are now adapter names; the endpoint code contains no direct agent/conversation SQL ownership.

Systematic consumption audit:

| Seam | Result |
| --- | --- |
| v1 `agent.list` | Moved to canonical store owner; supports `flow`/`role`. |
| v1 `agent.get` | Moved to canonical store owner; missing agent maps to `AGENT_NOT_FOUND`. |
| v1 `conversation.list` | Moved to canonical store owner; supports `agent_id`, `run_id`, `limit`, `cursor`. |
| v1 `conversation.get` | Moved to canonical store owner; missing session maps to `SESSION_NOT_FOUND`. |
| v1 `conversation.current_for_agent` | Moved to conversation owner; unknown agent maps to `AGENT_NOT_FOUND`, no active conversation returns `null`. |
| Dashboard `/api/agents*` | Moved to canonical owner as legacy DTO adapter. |
| Dashboard `/api/conversations*` | Moved to canonical owner as legacy DTO adapter. |
| Dashboard SQL reader files | No direct `agent_sessions` / `agent_turns` / `agent_conversation_audits` / `QueryContext` / `QueryRowContext` same-concept ownership remains in agent/conversation endpoint files. |
| Builder/CLI/script reads | No live same-concept interpreter found during audit; remains parent residual only for broader API migration. |
| Runtime session writers/fork/read-debug/entity/observability | Different semantic concepts or previously closed sibling owners; consumed as persisted facts only. |

Old non-authoritative paths invalidated:
- Dashboard handler fallback over `LoadAgents` no longer answers `/api/agents*`.
- Dashboard agent SQL projection is no longer local ownership; `SQLAgentReader` delegates to store owner.
- Dashboard conversation SQL projection is no longer local endpoint ownership; deprecated compatibility query paths live in the store owner and dashboard endpoints call canonical owner methods.

Sibling contexts checked:
- #696/#697 observability reads: not reopened.
- #700/#701 entity reads: not reopened.
- #677/#683 run/runtime control: out of scope.
- Slice 7 subscriptions and CLI migration: out of scope and still parent tail.

Manifestation coverage:

| Manifestation | Status | Exact proof |
| --- | --- | --- |
| `agent.list` unavailable | reproduced and fixed | `internal/apiv1/operator_agent_conversation_test.go` calls method and verifies store options/result. |
| `agent.get` unavailable / no `AGENT_NOT_FOUND` | reproduced and fixed | v1 typed-error test maps `store.ErrAgentNotFound` to `AGENT_NOT_FOUND`. |
| `conversation.list` unavailable / filters absent | reproduced and fixed | v1 test verifies `agent_id`, `run_id`, `limit`, `cursor`; store test verifies canonical projection query/result. |
| `conversation.get` unavailable / no `SESSION_NOT_FOUND` | reproduced and fixed | v1 typed-error test maps `store.ErrSessionNotFound` to `SESSION_NOT_FOUND`. |
| `conversation.current_for_agent` missing | reproduced and fixed | v1 test covers method dispatch through owner; store owner implements unknown-agent/null-current semantics. |
| Dashboard `/api/agents*` same-concept local projection | execution-proven through same corrected path | dashboard handlers require `canonicalAgentReader`; tests updated to prove adapter path. |
| Dashboard `/api/conversations*` same-concept local projection | execution-proven through same corrected path | dashboard handlers require `canonicalConversationReader`; SQL grep shows endpoint files no longer contain direct same-concept SQL ownership. |
| Task conversation visibility regression risk | execution-proven through same corrected path | store owner preserves `CanonicalTaskConversationVisibilitySourceSQL`; dashboard/conformance tests pass. |
| Malformed persisted conversation/turn payloads | execution-proven through same corrected path | existing dashboard/conversation fail-closed tests pass through the store-owned compatibility methods. |
| Builder/CLI hidden same-concept reads | split / escalated as separate class | audit found docs/historical pressure only, no live handler; parent #665/#694 remains open. |

Generic failing proof used or created:
- Created v1 agent/conversation handler tests.
- Created store owner conversation projection test.
- Preserved dashboard fail-closed tests for canonical turn/runtime-state decoding.

Required supported-surface proof run:
- `go run ./cmd/swarm-openrpc-gen --check` -> `api spec valid: methods=36 schemas=49 errors=21 mutating=12 subscriptions=3`.
- `go test ./internal/store ./internal/apiv1 ./internal/dashboard/server -count=1` -> passed.
- `go test ./... -count=1` -> passed.
- No-old-owner grep: `rg -n "agent_sessions|agent_turns|agent_conversation_audits|QueryContext|QueryRowContext" internal/dashboard/server/agents_sql.go internal/dashboard/server/conversations_sql.go internal/dashboard/server/server.go` -> no matches.

Closure level achieved: `failure class eliminated` for #704 only.

Same-concept interpreters still live: none in the approved #704 boundary. The remaining same-family work is outside this child and remains tracked under #665/#694.

Watchlist decision: no watchlist file change needed; work maps to existing `operator-surfaces.yaml` nodes `typed_persisted_operator_reads` and `canonical_operator_projection_truth`.

Deeper architecture/type-model smell: dashboard still carries legacy DTO names during the deprecation window. Long-run direction is to retire dashboard REST DTO compatibility and consume v1 DTOs directly. Tracking decision: parent residual under #665/#694, not closure-bearing for #704.

Optional general feedback: none closure-bearing beyond the parent tails already tracked.

File: internal/store/operator_agent_conversation_read_surface.go [hunk 1]
```diff
@@ -0,0 +1,1655 @@
...
+		SessionID:           strings.TrimSpace(projection.SessionID),
+		ProviderSessionID:   strings.TrimSpace(projection.ProviderSessionID),
+		CurrentTaskID:       strings.TrimSpace(projection.CurrentTaskID),
+		LastTool:            projection.LastTool,
+		LiveTurn:            projection.LiveTurn,
+		StartedAt:           row.StartedAt,
+		DashboardStatus:     strings.TrimSpace(projection.Status),
+		DashboardState:      projection.dashboardState(),
+		BlockingLayer:       projection.dashboardBlockingLayer(),
+		CurrentSessionRef:   projection.currentSessionRef(),
```
Comment (yazzaoui) [inline internal/store/operator_agent_conversation_read_surface.go:1011]: Blocking: this path still drops the `agent.get` refs required by the spec. `projection.currentSessionRef()` depends on `SessionStartedAt`, but `loadAgentOperatorProjections` never selects or scans the session `created_at`, so it always returns nil; `projection.LastTurnRef` is also never assigned from the latest turn. The #704 gate required `agent.get` to return refs to the current session and last turn, not just route through a shared owner.

File: internal/store/operator_agent_conversation_read_surface.go [hunk 2]
```diff
@@ -0,0 +1,1655 @@
...
+
+func (r *OperatorAgentConversationReadSurface) LoadCurrentOperatorConversationForAgent(ctx context.Context, agentID string) (*OperatorConversationDetail, error) {
+	agentID = strings.TrimSpace(agentID)
+	if agentID == "" {
+		return nil, ErrAgentNotFound
+	}
+	if _, err := r.LoadOperatorAgent(ctx, agentID); err != nil {
+		return nil, err
+	}
+	result, err := r.ListOperatorConversations(ctx, OperatorConversationListOptions{AgentID: agentID, Limit: 1})
```
Comment (yazzaoui) [inline internal/store/operator_agent_conversation_read_surface.go]: Blocking: `conversation.current_for_agent` must return the agent's most-recent active session, or null if there is no active session. This delegates to the generic conversation list with only `AgentID`/`Limit`, whose sources include terminated `agent_sessions` and task audit rows and sort by latest update. A terminated/audit conversation can therefore be returned as the current session instead of null.

**Review by yazzaoui — COMMENTED** [review]
Changes needed. I verified the PR description, existing PR discussion, linked issue #704, the pre-implementation audit/gate, parent trackers #665/#694, and the cited `platform-spec.yaml` sections for `agent.get` and `conversation.current_for_agent`. The issue category/boundary is acceptable for the approved #704 child, and the dashboard/API routing mostly moves toward the shared store owner, but the implementation does not eliminate the chosen failure class.

Blocking findings:

1. `agent.get` still cannot return the current session ref or last turn ref required by the spec/proof audit. The projection has those fields, but the SQL never selects/scans the session start time and never assigns `LastTurnRef`, so `current_session_ref` and `last_turn_ref` are silently omitted even when an active session/latest turn exists.

2. `conversation.current_for_agent` returns the latest conversation row for the agent, not the most-recent active live session. The generic list source includes terminated sessions and task audit rows, and the call does not filter by active/live-session semantics. That can return historical/audit data instead of `null` when there is no active session, contrary to the spec.

These are not follow-up-class issues; they are in-scope #704 manifestations and need code/test fixes in this PR before merge.

**Comment by yazzaoui** [comment]
PR review checklist - changes needed

- Guideline/spec checks: fail. `platform-spec.yaml` requires `agent.get` to expose current-session/last-turn refs and `conversation.current_for_agent` to return the most-recent active session or null.
- Failure-class coverage: fail. The #704 class is not eliminated while those two closure-bearing manifestations are still wrong in the canonical owner.
- Concept-wide drift reduction: partial. v1 and dashboard routes now point at the shared store owner, and I did not find a surviving dashboard-local SQL owner for the approved #704 surfaces, but the owner itself still encodes incorrect semantics.
- Closure level: not achieved; below failure-class eliminated.
- Tests run by reviewer: `go run ./cmd/swarm-openrpc-gen --check`; `go test ./internal/store ./internal/apiv1 ./internal/dashboard/server -count=1`.
- Residual risk: high until store-owner tests assert non-nil `agent.get` refs from active session/latest turn data and `conversation.current_for_agent` excludes terminated sessions/task audits and returns null with no active live session.
- Watchlist/tracker decision: no new issue/watchlist change. This is in-scope #704 work under existing #665/#694 and existing operator-surfaces watchlist nodes.
- Merge recommendation: do not merge until the two blocking findings from review `4269000578` are fixed and proven.

File: internal/store/operator_agent_conversation_read_surface.go (hunk 1 above)
Comment (yazzaoui) [reply internal/store/operator_agent_conversation_read_surface.go:1011]: Fixed in `e226d791`.

`loadAgentOperatorProjections` now selects/scans the active session `created_at` and latest turn `error`/`created_at`, then populates `CurrentSessionRef` and `LastTurnRef` from the canonical store projection. Added `TestOperatorAgentReadSurfaceLoadAgentProjectsSessionAndTurnRefs` to assert non-nil refs from active session/latest turn data.

Proof run:
- `go test ./internal/store -run 'TestOperator(AgentReadSurfaceLoadAgentProjectsSessionAndTurnRefs|ConversationReadSurfaceCurrentForAgent)' -count=1`
- `go run ./cmd/swarm-openrpc-gen --check`
- `go test ./internal/store ./internal/apiv1 ./internal/dashboard/server -count=1`

File: internal/store/operator_agent_conversation_read_surface.go (hunk 2 above)
Comment (yazzaoui) [reply internal/store/operator_agent_conversation_read_surface.go]: Fixed in `e226d791`.

`conversation.current_for_agent` no longer delegates to the generic conversation list. It now uses a live-session-only owner path that queries `agent_sessions` with `status = 'active'`, session runtime modes only, ordered by `updated_at DESC, created_at DESC`, and returns `nil` when no active live session exists. Terminated sessions and task-audit rows are not part of this current-session query.

Added store-owner tests for both the most-recent-active path and the null-without-active-live-session path.

Proof run:
- `go test ./internal/store -run 'TestOperator(AgentReadSurfaceLoadAgentProjectsSessionAndTurnRefs|ConversationReadSurfaceCurrentForAgent)' -count=1`
- `go run ./cmd/swarm-openrpc-gen --check`
- `go test ./internal/store ./internal/apiv1 ./internal/dashboard/server -count=1`

**Comment by yazzaoui** [comment]
Review follow-up pushed in `e226d791`.

Addressed the two blocking findings:
- `agent.get` now returns `current_session_ref` and `last_turn_ref` from the canonical store projection.
- `conversation.current_for_agent` now uses active live sessions only and returns `null` when none exists.

Added store-owner tests for both closure points and updated dashboard SQL mocks for the widened projection.

Proof run:
- `go test ./internal/store -run 'TestOperator(AgentReadSurfaceLoadAgentProjectsSessionAndTurnRefs|ConversationReadSurfaceCurrentForAgent)' -count=1`
- `go test ./internal/dashboard/server -run 'TestSQLAgentReader_' -count=1`
- `go run ./cmd/swarm-openrpc-gen --check`
- `go test ./internal/store ./internal/apiv1 ./internal/dashboard/server -count=1`

**Review by yazzaoui — COMMENTED** [review]
Re-review at `e226d791`.

No blocking findings remain. I re-read the PR description, PR discussion, inline review threads, issue #704, the #704 pre-implementation audit and gate outcome, parent trackers #665/#694, and the cited `platform-spec.yaml` sections for `agent.list`, `agent.get`, `conversation.list`, `conversation.get`, `conversation.current_for_agent`, `AgentSummary`, `AgentDetail`, `ConversationSummary`, and `ConversationDetail`.

The prior blocking findings are verified obsolete on current head:

- `agent.get` now populates `current_session_ref` and `last_turn_ref` from the canonical store projection. `loadAgentOperatorProjections` selects/scans active session `created_at` and latest-turn completion/error fields, and `TestOperatorAgentReadSurfaceLoadAgentProjectsSessionAndTurnRefs` proves non-nil refs from seeded active session/latest turn data.
- `conversation.current_for_agent` no longer delegates to generic conversation listing. It uses a live-session-only query over active session runtime modes and returns `nil` when no active live session exists; the store tests cover both most-recent-active and null-without-active-session behavior.

Code-level adversarial checks matched the audit claims. The v1 handlers and dashboard adapter methods consume the shared store owner, and the no-old-owner grep for direct dashboard agent/conversation SQL ownership returned no matches in `internal/dashboard/server/agents_sql.go`, `internal/dashboard/server/conversations_sql.go`, and `internal/dashboard/server/server.go`. The remaining `SQLAgentReader` / `SQLConversationReader` names are adapter wrappers, not independent SQL owners.

I treated the `e226d791` review-follow-up comments as the proof-audit supplement for the two previously missed manifestations; with that supplement, all currently known #704 manifestations have code-level proof. Parent tails remain tracked under #665/#694 and are not part of this closure claim.

Merge recommendation: merge. Closure level is `failure class eliminated` for #704 only.

**Comment by yazzaoui** [comment]
PR review checklist - merge recommended

- Guideline/spec checks: pass. Current head `e226d791` matches the cited `platform-spec.yaml` method/schema contracts for the five #704 read methods.
- Failure-class coverage: pass for #704. The shared store owner now covers v1 `agent.*` / `conversation.*` reads and dashboard REST is adapter-only for the approved child boundary.
- Concept-wide drift reduction: pass. I found no surviving dashboard-local same-concept SQL owner for the approved agent/conversation read surfaces.
- Prior review threads: resolved after verification on current head. Both earlier blockers are code-fixed and test-covered.
- Closure level: `failure class eliminated` for #704 only. #665/#694 remain open for subscriptions, CLI/API migration, agent controls, and other parent tails.
- Tests run by reviewer: `go test ./internal/store -run 'TestOperator(AgentReadSurfaceLoadAgentProjectsSessionAndTurnRefs|ConversationReadSurfaceCurrentForAgent)' -count=1`; `go run ./cmd/swarm-openrpc-gen --check`; `go test ./internal/store ./internal/apiv1 ./internal/dashboard/server -count=1`; `go test ./internal/dashboard/server -run 'TestSQLAgentReader_|TestSQLConversationReader_' -count=1`; `go test ./... -count=1`.
- Residual risk: low for #704. Dashboard legacy DTO compatibility remains a parent architecture/deprecation tail, already captured by #665/#694.
- Watchlist/tracker decision: no new issue or watchlist change. Existing operator-surfaces nodes and #665/#694 still cover the remaining non-#704 work.
- Merge recommendation: merge.
